app/controllers/items_controller.rb

```diff
@@ -690,9 +687,9 @@ def add_workflow
 
     # We need to sync up the workflows datastream with workflow service (using #find)
     # and then force a committed Solr update before redirection.
-    reindex Dor::Item.find(params[:id])
-    msg = "Added #{wf_name}"
+    reindex find_druid(params[:id])
```

Comment:
`Dor::Item.find` and `Dor.find` do not follow the same logic, so calling `find_druid` (which uses `Dor.find`) in this case changes the behavior.